sklearn/metrics/classification.py

```diff
@@ -185,6 +185,68 @@ def accuracy_score(y_true, y_pred, normalize=True, sample_weight=None):
     return _weighted_sum(score, sample_weight, normalize)
 
 
+def multilabel_confusion_matrix(y_true, y_pred):
+    """Compute True positive, False positive, True negative, False negative
+    for a multilabel classification problem.
+
+    Parameters
+    ----------
+    y_true : array, shape = (n_samples, n_labels)
+        Ground truth (correct) target values.
+
+    y_pred : array, shape = (n_samples, n_labels)
+        Estimated targets as returned by a classifier.
+
+    Returns
+    -------
+    C : ndarrays of shape = (n_labels, 4)
+        The multilabel confusion matrix.
+
+        The labels are represented by rows and the coloumns represent
+        the various confusion values in the following order - tp, fp, tn and fn
+
+        C[i, :] returns the various confusion values for label i and C[:, 0]
+        returns the true positive values for all labels.
+
+    References
+    ----------
+    .. [1] `Wikipedia entry for the Confusion matrix
+           <http://en.wikipedia.org/wiki/Sensitivity_and_specificity>`_
+       [2] http://www.cnts.ua.ac.be/~vincent/pdf/microaverage.pdf
+
+    Examples
+    --------
+    >>> from sklearn.metrics import multilabel_confusion_matrix
+    >>> y_true = np.array([[1, 0, 0], [1, 0, 0]])
+    >>> y_pred = np.array([[0, 1, 1], [1, 0, 0]])
+    >>> multilabel_confusion_matrix(y_true, y_pred)[:, 0]
+    array([1, 0, 0])
+    """
+    y_type, y_true, y_pred = _check_targets(y_true, y_pred)
+    if y_type != 'multilabel-indicator':
+        raise ValueError("%s is not supported" % y_type)
+
+    n_labels = y_true.get_shape()[1]
+
+    mcc = np.empty(shape=(n_labels, 4), dtype=int)
+
+    for label_idx in range(n_labels):
+        y_pred_col = y_pred.getcol(label_idx)
```

Comment:
I can't remember the state of things, but I think this needs to support dense or sparse `y` and you should use the `count_nonzero` helper.